sdk/core/azure-json-gson/src/test/java/com/azure/json/gson/GsonJsonWriterContractTests.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.json.gson;
+
+import com.azure.json.JsonWriter;
+import com.azure.json.contract.JsonWriterContractTests;
+import com.azure.json.implementation.AccessibleByteArrayOutputStream;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Tests {@link GsonJsonWriter} against the contract required by {@link JsonWriter}.
+ */
+public class GsonJsonWriterContractTests extends JsonWriterContractTests {
```

Comment:
not for this PR, it'd be nice to have a set of common tests that any json implementation can run against